### PR TITLE
Godhand Rework; The More Competant Version

### PR DESCRIPTION
**Pull request by TheKillfish**
## What Does This PR Do
Revives my original, janky rework for Godhand (#22575) and redone it with recommended changes and greater coding competancy. This PR does _not_ include new sprites, those can and will be done later.

## Why It's Good For The Game
Much the same as original; Godhand is thematically cool but doesn't do much or as much as it could. Cool, you have the equivalent of a lit welder permanently glued to your hand that does nothing else. This lets you light most flammable things on fire and emits passive light, all the while _not_ being an active fire hazard (so no need to fear plasmafloods).

## Testing
Tweaked and relocated code, smoked a lot of cigarettes, burned a lot of paper, burned a lot of vodka.

## Changelog
:cl:
tweak: Made Godhand mechanically more interesting
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/holy_weapons.dm [hunk 1]
```diff
@@ -120,7 +120,40 @@
...
+
+/obj/item/nullrod/godhand/proc/update_brightness()
+	set_light(2)
+
+/obj/item/nullrod/godhand/get_heat()
+	return 1500
+
+/obj/item/nullrod/godhand/attack(mob/living/carbon/M, mob/user)
+	if(HAS_TRAIT(user, TRAIT_PACIFISM))
+		to_chat(user, "<span class='warning'>You don't want to harm other living beings!</span>")
```
Comment (DGamerL) [inline code/game/objects/items/weapons/holy_weapons.dm:139]: You have to actually return this

**Comment by Henri215** [comment]
Remember to update your changelog later for a more descriptive one, players will have 0 idea on what interesting mechanics are.

File: code/game/objects/items/weapons/holy_weapons.dm [hunk 2]
```diff
@@ -120,7 +120,40 @@
...
+	if(!isliving(M) || !ismob(M))
+		return
+	M.IgniteMob()
```
Comment (DGamerL) [inline code/game/objects/items/weapons/holy_weapons.dm:142]: So it'll ignite someone even if you only want to light their cig?

File: code/game/objects/items/weapons/holy_weapons.dm [hunk 3]
```diff
@@ -120,7 +120,40 @@
...
+			cig.attackby(src, user)
+		else
+			if(istype(src, /obj/item/nullrod/godhand))
+				cig.light("<span class='rose'>[user] extends [user.p_their()] hand out to [M], and [cig] ignites.</span>")
+			else
+				cig.light("<span class='notice'>[user] reaches out to [M] with [user.p_their()] hand, and [cig] suddenly ignites.</span>")
+			playsound(src, 'sound/items/lighter/light.ogg', 25, TRUE)
+			M.update_inv_wear_mask()
+	else
+		..()
```
Comment (DGamerL) [inline code/game/objects/items/weapons/holy_weapons.dm:156]: Rewrite this using a guard clause please

**Review by DGamerL — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/holy_weapons.dm (hunk 2 above)
Comment (TheKillfish) [reply code/game/objects/items/weapons/holy_weapons.dm:142]: Same code is present with normal lighters, so keeping it for consistency

Comment (DGamerL) [reply code/game/objects/items/weapons/holy_weapons.dm:142]: That might be subject to change in the future aswell, no need for consistency if it is weird.
If you think that the godhand should ignite people when you light a cig for them, keep it in, if not, change it

File: code/game/objects/items/weapons/holy_weapons.dm (hunk 3 above)
Comment (TheKillfish) [reply code/game/objects/items/weapons/holy_weapons.dm]: Hasn't been changed to a full guard clause yet, but has been cut down

**Comment by TheKillfish** [comment]
I think this is fine enough to undraft

File: code/game/objects/items/candle.dm [hunk 4]
```diff
@@ -43,8 +43,12 @@
 
 /obj/item/candle/attackby(obj/item/W, mob/user, params)
 	if(W.get_heat())
-		light("<span class='notice'>[user] lights [src] with [W].</span>")
-		return
+		if(istype(W, /obj/item/nullrod/godhand))
+			light("<span class='rose'>[user] holds their hand over the wick of [src] and it starts burning.</span>")
```
Comment (lewcc) [inline code/game/objects/items/candle.dm:47]:
```suggestion
			light("<span class='rose'>[user] holds [user.p_their()] hand over the wick of [src] and it starts burning.</span>")
```

File: code/game/objects/items/weapons/cigs.dm [hunk 5]
```diff
@@ -138,6 +138,9 @@ LIGHTERS ARE IN LIGHTERS.DM
 				explosion(user.loc, -1, 0, 2, 3, 0, flame_range = 2)
 			F.charges--
 
+	else if(istype(I, /obj/item/nullrod/godhand))
+		light("<span class='rose'>[user] calmly raises [user.p_their()] hand to their [name], and it suddenly ignites.</span>")
```
Comment (lewcc) [inline code/game/objects/items/weapons/cigs.dm:142]:
```suggestion
		light("<span class='rose'>[user] calmly raises [user.p_their()] hand to [user.p_their()] [name], and it suddenly ignites.</span>")
```

File: code/game/objects/items/weapons/holy_weapons.dm [hunk 6]
```diff
@@ -120,7 +120,40 @@
...
+
+/obj/item/nullrod/godhand/attack(mob/living/carbon/M, mob/user)
+	if(HAS_TRAIT(user, TRAIT_PACIFISM))
+		to_chat(user, "<span class='warning'>You don't want to harm other living beings!</span>")
+		return
+	if(!isliving(M))
+		return
+	M.IgniteMob()
+	if(!ismob(M))
+		return
```
Comment (lewcc) [inline code/game/objects/items/weapons/holy_weapons.dm:145]: Given that you've already done isliving (which checks for `mob/living`) you probably don't need this check here

File: code/game/objects/items/weapons/holy_weapons.dm [hunk 7]
```diff
@@ -120,7 +120,40 @@
...
+
+	if(istype(M.wear_mask, /obj/item/clothing/mask/cigarette) && user.zone_selected == "mouth")
```
Comment (lewcc) [inline code/game/objects/items/weapons/holy_weapons.dm:147]:
```suggestion
	if(istype(M.wear_mask, /obj/item/clothing/mask/cigarette) && user.zone_selected == BODY_ZONE_PRECISE_MOUTH)
```

File: code/game/objects/items/candle.dm [hunk 8]
```diff
@@ -43,8 +43,12 @@
 
 /obj/item/candle/attackby(obj/item/W, mob/user, params)
 	if(W.get_heat())
```
Comment (DGamerL) [inline code/game/objects/items/candle.dm:45]: This can probably just become a guard clause now

**Comment by TheKillfish** [comment]
Oh goddamnit, those weren't sposed to be merged. Drafting this to sort out this technical difficulty

**Pull request closed without merging**